pkg/util/ipvs/ipvs.go

```diff
@@ -0,0 +1,92 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package ipvs
+
+import (
+	"net"
+	"strconv"
+)
+
+// Interface is an injectable interface for running ipvs commands.  Implementations must be goroutine-safe.
+type Interface interface {
+	// Flush clears all services in system. return occurred error immediately.
+	Flush() error
+	// EnsureDummyDevice checks if the specified dummy interface is present and, if not, creates it.  If the dummy interface existed, return true.
+	EnsureDummyDevice(dummyDev string) (exist bool, err error)
+	// DeleteDummyDevice deletes the specified dummy interface.  If the dummy interface existed, return error.
```

Comment:
I think if the dummy interface exist, DeleteDummyDevice should delete it instead of returning error.